UMLEditor/UMLEditor/Views/CommandLine.cs

```diff
@@ -61,54 +65,347 @@ private void ExecuteCommand(string input)
 
                 break;
 
+            
             case ("delete_class"):
-                //TODO 
-                break;
+                if (arguments.Count == 1)
+                {
+                    // Try to Delete a Class with the first provided word as its name
+                    _activeDiagram!.DeleteClass(arguments[0]);
 
+                    PrintColoredLine($"Class {arguments[0]} deleted.", SUCCESS_COLOR);
+                }
+
+                else
+                {
+                    PrintColoredLine("To delete an existing Class, please enter \"delete_class\" " +
+                                     "followed by a class name into the console and then press enter.", ERROR_COLOR);
+                }
+                break;
+                
             case ("rename_class"):
-                //TODO 
+                if (arguments.Count == 2)
+                {
+                    // Try to Rename a Class with the first provided word as its name
+                    _activeDiagram!.RenameClass(arguments[0],  arguments[1]);
+
+                    PrintColoredLine($"Class {arguments[0]} renamed to {arguments[1]}", SUCCESS_COLOR);
+                }
+
+                else
+                {
+                    PrintColoredLine("To rename a Class, please enter \"rename_class\" " +
+                                     "followed by a class NewName and class OldName into the console and then press enter.", ERROR_COLOR);
+                }
                 break;
 
             case ("add_field"):
-                //TODO 
+                if (arguments.Count == 3)
+                {
+                    Class? currentClass = _activeDiagram!.GetClassByName(arguments[0]);

```

Comment:
There's some encroaching on the model by the view going on here.  We should look into having stuff be done within the diagram so that GetClassByName can be private.